src/modules/effects/damage.cpp

```diff
@@ -0,0 +1,287 @@
+
+			TextureCount = 0;
+		}
+
+		if (VertexCount)
+		{
+			delete[] VertexPositions;
+			VertexPositions = nullptr;
+
+			VertexCount = 0;
+		}
+	}
+
+	void mmDamage::Init(modStatic* model, modShader* shaders, int shaderCount)
+	{
+		if (model == nullptr)
+			return;
+
+		BodyModel = model;
+		CleanModel = model->Clone();
+		DamageModel = model->Clone();
+
+		TextureCount = shaderCount;
+		CleanShaders = new modShader[shaderCount];
+		DamageShaders = new modShader[shaderCount];
+		DamageTextures = new gfxTexture*[shaderCount];
+
+		for (int i = 0; i < shaderCount; i++)
+		{
+			CleanShaders[i].SetTexture(shaders[i].GetTexture());
+			CleanShaders[i].SetMaterial(shaders[i].GetMaterial());
+
+			DamageShaders[i].SetTexture(shaders[i].GetTexture());
+			DamageShaders[i].SetMaterial(shaders[i].GetMaterial());
+
+			DamageTextures[i] = nullptr;
+		}
+
+		bool hasDamageTextures = false;
+
+		for (int i = 0; i < CleanModel->GetPacketCount(); ++i)
+		{
+			int shaderIndex = CleanModel->GetShaderIndex(i);
+			modShader* cleanShader = &CleanShaders[shaderIndex];
+			cleanShader->PreLoad();
+
+			if (cleanShader->GetTexture())
+			{
+				char textureName[128];
+
+				strcpy_s(textureName, cleanShader->GetTexture()->GetName());
+
+				char* find = strrchr(textureName, '_');
+
+				if (find && !_strcmpi(find, "_dmg"))
+				{
+					*find = '\0';
+
+					gfxTexture* cleanTexture = gfxGetTexture(textureName, 1);
+
+					if (cleanTexture)
+					{
+						cleanShader->SetTexture(cleanTexture);
+						DamageTextures[shaderIndex] = cleanTexture;
+						hasDamageTextures = true;
+					}
+				}
+			}
+		}
+
+		for (int i = 0; i < DamageModel->GetPacketCount(); ++i)
+		{
+			int shaderIndex = DamageModel->GetShaderIndex(i);
+			modShader* damageShader = &DamageShaders[shaderIndex];
+			damageShader->PreLoad();
+
+			if (damageShader->GetTexture())
+			{
+				char textureName[128];
+
+				strcpy_s(textureName, damageShader->GetTexture()->GetName());
+
+				char* find = strrchr(textureName, '_');
+
+				if (!find || _strcmpi(find, "_dmg"))
+				{
+					strcat_s(textureName, "_dmg");
+
+					gfxTexture* damageTexture = gfxGetTexture(textureName, 1);
+
+					if (damageTexture)
+					{
+						damageShader->SetTexture(damageTexture);
+						DamageTextures[shaderIndex] = damageTexture;
+						hasDamageTextures = true;
+					}
+				}
+			}
+		}
+
+		if (hasDamageTextures)
+		{
+			int adjunctCount = model->GetAdjunctCount();
+
+			VertexCount = adjunctCount;
+
+			VertexPositions = new Vector3[adjunctCount];
+
+			int currentVertex = 0;
+
+			for (int i = 0; i < model->GetPacketCount(); i++)
+			{
+				auto packet = model->GetPacket(i);
+				while (packet)
+				{
+					packet->DoLock();
+
+					for (unsigned int j = 0; j < packet->GetAdjunctCount(); j++)
+					{
+						packet->GetPosition(VertexPositions[currentVertex], j);
+						currentVertex++;
+					}
+
+					packet->DoUnlock();
+					packet = packet->GetNext();
+				}
+			}
+		}
+
+		Reset(0);
```

Comment:
Since this is a boolean, change this to false for clarity

Reply:
Done!